packages/bitcoin-wallet-provider/lib/BitcoinWalletProvider.js

```diff
@@ -226,6 +236,102 @@ export default superclass => class BitcoinWalletProvider extends superclass {
       .then(({ unusedAddress }) => unusedAddress[key])
   }
 
+  async getInputsForSweep (_targets = [], _feePerByte, fixedInputs = [], numAddressPerCall = 100) {
+    let addressIndex = 0
+    let changeAddresses = []
+    let nonChangeAddresses = []
+    let addressCountMap = {
+      change: 0,
+      nonChange: 0
+    }
+
+    const feePerBytePromise = this.getMethod('getFeePerByte')()
+    let feePerByte = _feePerByte || false
+
+    while (addressCountMap.change < ADDRESS_GAP || addressCountMap.nonChange < ADDRESS_GAP) {
+      let addrList = []
+
+      if (addressCountMap.change < ADDRESS_GAP) {
+        // Scanning for change addr
+        changeAddresses = await this.getAddresses(addressIndex, numAddressPerCall, true)
+        addrList = addrList.concat(changeAddresses)
+      } else {
+        changeAddresses = []
+      }
+
+      if (addressCountMap.nonChange < ADDRESS_GAP) {
+        // Scanning for non change addr
+        nonChangeAddresses = await this.getAddresses(addressIndex, numAddressPerCall, false)
+        addrList = addrList.concat(nonChangeAddresses)
+      }
+
+      let utxos
+      if (fixedInputs.length === 0) {
+        utxos = await this.getMethod('getUnspentTransactions')(addrList)
+        utxos = utxos.map(utxo => {
+          const addr = addrList.find(a => a.equals(utxo.address))
+          return {
+            ...utxo,
+            value: BigNumber(utxo.amount).times(1e8).toNumber(),
+            derivationPath: addr.derivationPath
+          }
+        })
+      } else {
+        utxos = fixedInputs
+      }
+
+      const utxoBalance = utxos.reduce((a, b) => a + (b['value'] || 0), 0)
+
+      const transactionCounts = await this.getMethod('getAddressTransactionCounts')(addrList)
+
+      if (feePerByte === false) feePerByte = await feePerBytePromise
+      const minRelayFee = await this.getMethod('getMinRelayFee')()
+      if (feePerByte < minRelayFee) {
+        throw new Error(`Fee supplied (${feePerByte} sat/b) too low. Minimum relay fee is ${minRelayFee} sat/b`)
+      }
+
+      const outputBalance = _targets.reduce((a, b) => a + (b['value'] || 0), 0)
+
+      const amountToSend = utxoBalance - (feePerByte * (((_targets.length + 1) * 39) + (utxos.length * 153))) // todo better calculation
```

Comment:
Tests passed with `legacy`, `p2sh-segwit` and `bech32`

We know that all utxos spent through this method, will be a p2pkh or p2wpkh txs (since we're spending from the wallet and not from p2sh or p2wsh txs) 

Looking at: https://github.com/bitcoinjs/coinselect/blob/master/utils.js#L3 we can see that

```
var TX_INPUT_BASE = 32 + 4 + 1 + 4
var TX_INPUT_PUBKEYHASH = 107
var TX_OUTPUT_BASE = 8 + 1
var TX_OUTPUT_PUBKEYHASH = 25
```

So a pkh input is (32 + 4 + 1 + 4) + 107 = 148
and a pkh output is (8 + 1) + 25 = 34

I added a bit of padding (5 bytes each) to arrive at 39 and 153 respectively

Reply:
This looks good. Unfortunately fees in `coinselect` are pre segwit, so likely not much we can do here for now: https://github.com/bitcoinjs/coinselect/issues/53